Datagrid/ProxyQuery.php

```diff
@@ -63,17 +63,38 @@ public function execute(array $params = array(), $hydrationMode = null)
         // always clone the original queryBuilder
         $queryBuilder = clone $this->queryBuilder;
 
+        $rootAlias = current($queryBuilder->getRootAliases());
+
         // todo : check how doctrine behave, potential SQL injection here ...
         if ($this->getSortBy()) {
             $sortBy = $this->getSortBy();
             if (strpos($sortBy, '.') === false) { // add the current alias
-                $sortBy = $queryBuilder->getRootAlias().'.'.$sortBy;
+                $sortBy = $rootAlias.'.'.$sortBy;
             }
             $queryBuilder->addOrderBy($sortBy, $this->getSortOrder());
         } else {
             $queryBuilder->resetDQLPart('orderBy');
         }
 
+        /* By default, always add a sort on the identifier fields of the first
+         * used entity in the query, because RDBMS do not guarantee a
+         * particular order when no ORDER BY clause is specified, or when
+         * the field used for sorting is not unique.
+         */
+
+        $identifierFields = $queryBuilder
+            ->getEntityManager()
+            ->getMetadataFactory()
+            ->getMetadataFor(current($queryBuilder->getRootEntities()))
+            ->getIdentifierFieldNames();
+
+        foreach ($identifierFields as $identifierField) {
+            $queryBuilder->addOrderBy(
```

Comment:
For instance, imagine you have a sort on the date, but two people happen to post something at the same exact second. You don't want the request to return things in a different order when you press F5, and a good way to achieve that is to append a sort on a unique key